client/client.go

```diff
@@ -31,8 +39,23 @@ type userAgentTransport struct {
 }
 
 var (
-	concurrentConnections          = 200
-	throttlingAdditionalRetryCount = 3
+	concurrentConnections = 200
+)
+
+var (
+	// A regular expression to match the error returned by net/http when the
+	// configured number of redirects is exhausted. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	redirectsErrorRe = regexp.MustCompile(`stopped after \d+ redirects\z`)
+
+	// A regular expression to match the error returned by net/http when the
+	// scheme specified in the URL is invalid. This error isn't typed
+	// specifically so we resort to matching on the error string.
+	schemeErrorRe = regexp.MustCompile(`unsupported protocol scheme`)
+
+	retryWaitMin = 1 * time.Second
+	retryWaitMax = 60 * time.Second
+	retryMax     = 40
```

Comment:
could you make those retry configurations configurable, so that they could be tuned easily?

Reply:
good suggestion.

Reply:
added by reading env